packages/react-components/src/components/MessageThread.tsx

```diff
@@ -67,6 +67,37 @@ const getLatestChatMessage = (messages: (ChatMessage | SystemMessage | CustomMes
   return undefined;
 };
 
+/**
+ * Get the latest participant number from the message array.
+ *
+ * @param messages
+ */
+const getLatestParticipantNum = (messages: (ChatMessage | SystemMessage | CustomMessage)[]): number => {
+  for (let i = messages.length - 1; i >= 0; i--) {
+    const message = messages[i];
+    if (message.messageType === 'chat') {
+      return message.messageThreadParticipantCount ? message.messageThreadParticipantCount : 1;
+    }
+  }
+  return 1;
+};
+
+/**
+ * Get the latest read number for my last sent message from the message array.
+ *
+ * @param messages
+ */
+const getLatestReadNum = (messages: (ChatMessage | SystemMessage | CustomMessage)[]): number => {
+  for (let i = messages.length - 1; i >= 0; i--) {
+    const message = messages[i];
+    if (message.messageType === 'chat' && message.mine && message.status === 'seen') {
+      // first find latest message that belongs to me, return read number for that message
+      return message.readReceipts ? message.readReceipts.length + 1 : 1;
+    }
+  }
```

Comment:
The assumption we made and the logic here is a little bit complex and easy to be missed - we can't just return message.ReadReceipt number and say this is our read count - cause people wouldn't send readReceipt for each message - all we need to do is to find one person's latest read line then determine whether a message has been read or not